scene/gui/file_dialog.cpp

```diff
@@ -649,6 +649,7 @@ void FileDialog::clear_filters() {
 }
 
 void FileDialog::add_filter(const String &p_filter) {
+	ERR_FAIL_COND_MSG(p_filter.begins_with("."), "Filter must be \"filename.extension\", can't start with dot.");
```

Comment:
Would there be a possibility that users would want to filter on hidden files starting with `.`?

Reply:
I think it's rather corner-case. And as I commented in the issue, filters starting with `.` don't work as expected. Supporting hidden files would require fixing that.